### PR TITLE
Use nnpdf single-truth theory

### PR DESCRIPTION
**Pull request by scarlehoff**
Changes to use https://github.com/NNPDF/nnpdf/pull/2028 with pineko.

At the moment it doesn't do anything clever, it just loads the yaml from there. I've tested the eko and fktable generation.
(the tests and so won't pass until the PR is merged in nnpdf)

### CODE REVIEW

**Comment by scarlehoff** [comment]
I've made a change (wrt #146). Now if the template and the theory have a different `mu0` this raises an Exception instead of overwriting the value silently.
I can revert the change back of course, but I think this is something important enough to be fail as early as possible (and the solution is "simply" to change the template or theory if one really wants to play god)

File: src/pineko/evolve.py [hunk 1]
```diff
@@ -122,8 +122,12 @@ def write_operator_card(pineappl_grid, default_card, card_path, tcard):
     xif = 1.0 if sv_method is not None else tcard["XIF"]
     # update scale variation method
     operators_card["configs"]["scvar_method"] = sv_method
-    # update initial scale mu0
+
+    # Make sure that we are using the theory Q0 and fail if the template has a different oen
```
Comment (felixhekhorn) [inline src/pineko/evolve.py:126]:
```suggestion
    # Make sure that we are using the theory Q0 and fail if the template has a different one
```

**Comment by scarlehoff** [comment]
One problem here is the nfonnl part.
However, I believe that, since now one can run nfonll by providing only the main theory id, the only logical thing is that the theory cards can also be autogenerated in the go (and so tcards should not be necessary).

This helps reducing the extra amount of theory cards and having to tell NNPDF about things like `PTOEKO` or `FONLLParts`.

**Comment by scarlehoff** [comment]
With the last commit from here https://github.com/NNPDF/nnpdf/pull/2067
this works with pineko to create theories without having to provide a theory or yamldb folder.

File: docs/source/overview/prerequisites.rst [hunk 2]
```diff
@@ -18,35 +18,47 @@ This is a standard example:
...
 
 
 All the relevant inputs are described below. The command ``pineko scaffold new`` will generate all necessary folders.
 
+
+Theory Runcards
+---------------
+
+You need to provide the necessary theory runcards named with their respective theory ID inside the *paths.theory_cards* folder [1]_ unless you are using the ``nnpdf`` key (see below).
+For more details about theory runcards you can look at https://eko.readthedocs.io/en/latest/code/IO.html under **Theory Runcards**.
```
Comment (felixhekhorn) [inline docs/source/overview/prerequisites.rst:43]: I'm not sure how much that link is up-to-date (the answer is it isn't) and it is surely not the official source of truth of which we are speaking here - instead it is NNPDF, i.e. https://docs.nnpdf.science/theory/theoryparamsdefinitions.html should be linked instead

Comment (scarlehoff) [inline docs/source/overview/prerequisites.rst:43]:
```suggestion
For more details about theory runcards you can look at https://docs.nnpdf.science/theory/theoryparamsdefinitions.html
```

File: docs/source/overview/running.rst [hunk 3]
```diff
@@ -95,3 +95,37 @@ With the command ``pineko compare`` it is possible to compare the predictions as
 
 again eventually specifying the values of *renormalization* and *factorization* scales with the
 appropriate options.
+
+Using pineko with NNPDF
+"""""""""""""""""""""""
+
+It is possible to use ``pineko`` without providing a mapping between data and grids
```
Comment (giacomomagni) [inline docs/source/overview/running.rst:102]: well this mapping is just hidden in the nnpdf metadata...

File: src/pineko/theory.py [hunk 4]
```diff
@@ -330,8 +330,11 @@ def eko(self, name, _grid, tcard):
         # from the PTO needed for the PDF evolution (and so by EKO). Here we
         # ensure that the PTO used in the EKO calculation reflects the real
         # perturbative order of the prescription.
-        if "PTOEKO" in tcard:
+        if tcard.get("PTOEKO") is not None:
             tcard["PTO"] = tcard["PTOEKO"]
+        # Keys still needed by eko below
```
Comment (giacomomagni) [inline src/pineko/theory.py:335]:
```suggestion
        # Deprecated keys still needed by eko below. TODO: remove them asap.
```

**Review by giacomomagni — APPROVED** [review]
I left two two useless comments.

**Review by felixhekhorn — APPROVED** [review]
I dropped most of the yamldb stuff, because it is not used here and nowhere else

File: docs/source/overview/running.rst [hunk 5]
```diff
@@ -95,3 +95,37 @@ With the command ``pineko compare`` it is possible to compare the predictions as
...
+(i.e., without a ``yamldb`` database) or theory cards, by using the data declared in the NNPDF
+repository.
```
Comment (scarlehoff) [inline docs/source/overview/running.rst:104]:
```suggestion
It is possible to use ``pineko`` without providing an explicit mapping between data and grids
(i.e., without a ``yamldb`` database) or theory cards, by using the data declared in the NNPDF
repository (which includes a data-theory mapping).
```

File: docs/source/overview/running.rst [hunk 6]
```diff
@@ -95,3 +95,37 @@ With the command ``pineko compare`` it is possible to compare the predictions as
...
 appropriate options.
+
+Using pineko with NNPDF
+"""""""""""""""""""""""
+
+It is possible to use ``pineko`` without providing an explicit mapping between data and grids
+(i.e., without a ``yamldb`` database) or theory cards, by using the data declared in the NNPDF
+repository (which includes a data-theory mapping).
+
+In order to do this you need to install ``pineko`` with the ``nnpdf`` extra, which will install
```
Comment (felixhekhorn) [inline docs/source/overview/running.rst:106]:
```suggestion
In order to do this you need to install ``pineko`` with the ``nnpdf`` extra, which installs
```
no need for future here, I would say - it is doing it right now

**Pull request merged**